CMakeTests.txt

```diff
@@ -0,0 +1,228 @@
+            target_include_directories(${name} PUBLIC
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src/apps/applib/include>)
+            target_compile_options(${name} PRIVATE ${H3_COMPILE_FLAGS})
+            target_link_libraries(${name} PRIVATE ${H3_LINK_FLAGS})
+        endif()
+
+        math(EXPR test_number "${test_number}+1")
+
+        add_test(NAME ${name}_test${test_number} COMMAND ${TEST_WRAPPER} "$<TARGET_FILE:${name}>")
+
+        if(ENABLE_COVERAGE)
+            add_custom_target(${name}_coverage${test_number}
+                COMMAND ${name} > /dev/null
+                COMMENT "Running ${name}_coverage${test_number}"
+                )
+
+            add_dependencies(coverage ${name}_coverage${test_number})
+            add_dependencies(${name}_coverage${test_number} clean-coverage)
+        endif()
+    endmacro()
+
+    macro(add_h3_test_common name srcfile)
+        # need to actually make the test target
+        if(NOT TARGET ${name})
+            add_h3_executable(${name} ${srcfile} ${APP_SOURCE_FILES} ${TEST_APP_SOURCE_FILES})
+        endif()
+
+        math(EXPR test_number "${test_number}+1")
+    endmacro()
+
+    macro(add_h3_test name srcfile)
+        add_h3_test_common(${name} ${srcfile})
+        add_test(NAME ${name}_test${test_number} COMMAND ${TEST_WRAPPER} "$<TARGET_FILE:${name}>")
+
+        if(ENABLE_COVERAGE)
+            add_custom_target(${name}_coverage${test_number}
+                COMMAND ${name} > /dev/null
+                COMMENT "Running ${name}_coverage${test_number}"
+                )
+
+            add_dependencies(coverage ${name}_coverage${test_number})
+            add_dependencies(${name}_coverage${test_number} clean-coverage)
+        endif()
+    endmacro()
+
+    macro(add_h3_test_with_file name srcfile argfile)
+        add_h3_test_common(${name} ${srcfile})
+        # add a special command (so we don't need to read the test file from the test program)
+        set(dump_command "cat")
+
+        add_test(NAME ${name}_test${test_number}
+                 COMMAND ${SHELL} "${dump_command} ${argfile} | ${TEST_WRAPPER_STR} $<TARGET_FILE:${name}>")
+
+        if(PRINT_TEST_FILES)
+            message("${name}_test${test_number} - ${argfile}")
+        endif()
+
+        if(ENABLE_COVERAGE)
+            add_custom_target(${name}_coverage${test_number}
+                COMMAND ${name} < ${argfile} > /dev/null
+                COMMENT "Running ${name}_coverage${test_number}"
+                )
+
+            add_dependencies(coverage ${name}_coverage${test_number})
+            add_dependencies(${name}_coverage${test_number} clean-coverage)
+        endif()
+    endmacro()
+
+    macro(add_h3_cli_test name h3_args expect_string)
+        add_test(NAME ${name}_test${test_number}
+          COMMAND ${SHELL} "test \"`$<TARGET_FILE:h3_bin> ${h3_args}`\" = '${expect_string}'")
+
+        if(PRINT_TEST_FILES)
+            message("${name}_test${test_number} - ${expect_string}")
```

Comment:
I'd suggest adding the `h3_args` here, as the purpose of PRINT_TEST_FILES is to connect which test number (which can be opaque) corresponds with which test in this file.

Reply:
```suggestion
            message("${name}_test${test_number} - ${h3_args} - ${expect_string}")
```